specification/compute/resource-manager/Microsoft.Compute/stable/2021-07-01/examples/communityGallery/GetACommunityGallery.json

```diff
@@ -0,0 +1,19 @@
+{
+  "parameters": {
+    "subscriptionId": "{subscription-id}",
+    "location": "myLocation",
+    "api-version": "2021-07-01",
+    "publicGalleryName": "publicGalleryName"
+  },
+  "responses": {
+    "200": {
+      "body": {
+        "location": "myLocation",
+        "name": "publicGalleryName",
+        "identifier": {
+          "uniqueId": "/CommunityGalleries/publicGalleryName"
+        }
+      }
```

Comment:
The response content should conform to ARM resource schema. (type, name, id properties are missing)
[https://armwiki.azurewebsites.net/api_contracts/RPCstrongvalidation.html?q=strong#get-resource](https://armwiki.azurewebsites.net/api_contracts/RPCstrongvalidation.html?q=strong#get-resource)

Reply:
Name is already there, for id, to distinguish normal RBAC sharing with community gallery(community gallery id does not contain gallery owners's subscription and resource group and we do not want to leak that as well), we use a new field to represent id "identifier-uniqueId", it is the same id as the one user requested.

Reply:
Our service's consumer hope we use a new field to represent direct sharing/community gallery so they do not get confused. The contract was built since last year and it worked fine since then.

Reply:
Adding your own ID property is fine, but all ARM resource should have a ID property. It's required by RPC.
[https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/resource-api-reference.md#get-resource](https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/resource-api-reference.md#get-resource)

If you need to return with ID, you can consider adding POST action to get some metadata about a gallery resource. The response schema for POST is up to RP.